recipe/2024-02-terraform-providers.md

```diff
@@ -0,0 +1,190 @@
+# Title
+
+* **Status**: Pending
+* **Author**: @lakshmimsft
+
+## Overview
+
+As part of Terraform support in Radius, we currently support azurerm, kubernetes and aws providers when creating recipes. We pull credentials stored in UCP and setup and save provider configuration which is accessible to recipes.\
+This document  describes a proposal to support multiple providers, setup their provider configuration, which would be accessible to all recipes within an  environment.
+
+## Terms and definitions
+
+| Term     | Definition                                                                                                                                                                                                 |
+| -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Terraform Provider | A Terraform provider is a plugin that allows Terraform to interact with and manage resources of a specific infrastructure platform or service, such as AWS, Azure, or Google Cloud. |
+
+
+## Objectives
+
+**Reference for new type recipeConfig and handling of secrets:** [Design document for Private Terraform Repository](https://github.com/radius-project/design-notes/blob/3644b754152edc97e641f537b20cf3d87a386c43/recipe/2024-01-support-private-terraform-repository.md)
+
+> **Issue Reference:** <!-- (If appropriate) Reference an existing issue that describes the feature or bug. -->
+https://github.com/radius-project/radius/issues/6539
+
+### Goals
+
+1. Enable users to use terraform recipes with multiple providers (including and outside of Azure, Kubernetes, AWS).
+2. Enable users to use terraform recipes with multiple configurations for the same provider. 
+3. Enable provider configurations to be available for all recipies used in an environment.
+
+
+### Non goals
+1. Updates to Bicep Provider configuration.
+2. Authentication for providers hosted in private registries.
+3. Support for .terraformrc files.
```

Comment:
Can you say more about why (what is a `.terraformrc` file?)

Reply:
https://developer.hashicorp.com/terraform/cli/config/config-file

Reply:
Let's update the doc to include this reference, we don't expect people to come back and look at comments once this is merged :). Also, let's add why this is a non goal.

Reply:
updated